lib/Sema/TypeCheckAvailability.cpp

```diff
@@ -4507,6 +4532,8 @@ class ProblematicTypeFinder : public TypeDeclFinder {
     // already checked on the TypeRepr.
     if (Where.mustOnlyReferenceExportedDecls())
       TypeChecker::diagnoseDeclRefExportability(Loc, decl, Where);
+
+    diagnoseNonRuntimeProtocol(Loc, decl);
```

Comment:
Hmm, no, you want to trigger this from `checkTypeMetadataAvailabilityInternal`, I think.  This place is much broader and will catch every use of the type.

Reply:
@rjmccall IIUC any use of a non-runtime protocol internal to a module should be okay (if/because) the compiler can flatten the substitution map into the concrete type no? Would we be able to lower other use of NRP to AnyObject after type-checking to use the objc runtime instead of Swift's protocol witnesses?

Reply:
ObjC protocols (NRP or not) always use the ObjC runtime, so yes, any uses as a value should be fine. That's not "any use ... internal to a module", though; you can't (currently) reliably make `Array<NRP>` without having metadata for the NRP.